syntax = "proto3";
package com.adv.msg;
import "adv_base.proto";

// 用户心跳上报，服务端之间
message UserKeepAliveRequest
{
    int32   condid        = 1;    // 接入服务id
    int32   devicetype    = 2;    // 客户端设备类型
    string  deviceid      = 3;    // 设备唯一ID
    int32   uid           = 4;    // 用户ID
    int32   version       = 5;    // 客户端版本号
    int32   transid       = 32;   // 标示请求
}

message UserKeepAliveResult
{
    RetBase retbase       = 1;
    int32   condid        = 2;    // 接入服务id
    int32   devicetype    = 3;    // 客户端设备类型
    string  deviceid      = 4;    // 设备唯一ID
    int32   uid           = 5;    // 用户ID
    int32   version       = 6;    // 客户端版本号
    int32   transid       = 32;   // 标示请求
}



// 登录协议
message LoginRequest 
{
    string  token         = 1;    //登录的token
    int32   magicnum      = 2;    //固定值123456789
    int32   status        = 3;    //客户端状态
    int32   devicetype    = 4;    //客户端设备类型
    string  deviceid      = 6;    //设备唯一ID
    string  description   = 7;    //设备名称
    int32   condid        = 8;    //所在cond的id
    int32   loginseq      = 9;    // 登录带到后端的序列号
    int32   accounttype   = 10;   // (1.qq,2.weixin)
    string  account       = 11;
    string  passwd        = 12;
    int32   version       = 13;
    string  nickname      = 14;   // 昵称
    int32   gender        = 15;   // 性别，0为女，1为男
    string  province      = 16;   // 所在省份
    string  city          = 17;   // 所在城市
    string  country       = 18;   // 所在国家
    string  avatar        = 19;   // 头像链接地址
    string  desc          = 20;   // 描述信息
    string  email         = 21;
    string  mobile        = 22;
    int32   time          = 23;   // 登录时间戳
    int32   uid           = 24;   // 用户uid，内部使用
    int32   flag          = 25;   // 登录方式，是否强制登录
    int32   transid       = 32;   // 标示请求
}

message LoginResult
{     
    RetBase retbase       = 1;
    string  key           = 2; // 服务端生成的密钥
    int32   uid           = 3; // 用户id
    int32   loginseq      = 4; // 登录带到后端的序列号
    int32   accounttype   = 5;   // (1.qq,2.weixin)
    int32   time          = 6;   // 登录时间戳
    int32   transid       = 32;   // 标示请求
}

message LogoutRequest 
{
    int32   deviceType    = 1;    //客户端设备类型
    string  deviceid      = 2;    //设备唯一ID
    int32   conid         = 3;    //所在cond的id
    int32   transid       = 32;   // 标示请求
}

message LogoutResult
{       
    RetBase   retbase     = 1;
    int32   transid       = 32;   // 标示请求
}

// 任务相关协议
message TaskNotify
{
    int32    tasktype     = 1;
    int32   transid       = 32;   // 标示请求
}

message SyncTaskRequest
{
    int64    syncpoint       = 1;
    int32    synctype     = 2; // (1,拉下载任务;2.拉文章任务)
    int32    syncflag     = 3; // (1,拉新任务;2.拉历史任务)
    int32    synclimit    = 4;
    int32   transid       = 32;   // 标示请求
}

message SyncTaskResult
{
    RetBase retbase       = 1;
    repeated TaskInfo taskinfos    = 2;
    int32    synctype     = 3; // (1,拉新任务;2.拉历史任务)
    int32   continueflag  = 4;
    int64   maxtaskid     = 5;
    int32   transid       = 32;   // 标示请求
}

message ReportTaskRequest
{
    TaskInfo  taskinfo    = 1;
    string    reward      = 2;    // 给后端到dbp的请求使用的字段，奖励金额
    int32   transid       = 32;   // 标示请求
}

message ReportTaskResult
{
    RetBase retbase       = 1;
    int64   taskid        = 2;
    string    reward      = 3;    // 奖励的金额
    int32   done_step     = 4;    // 当前完成到第几步
    int32   tasktype      = 5;    // 任务类型
    int32   report_time   = 6;    // 上报时间
    int32   transid       = 32;   // 标示请求
}


// 收入相关
message GetIncomeRequest
{
    int32   transid       = 32;   // 标示请求
}

message GetIncomeResult
{
    RetBase retbase       = 1;
    string  pre_cash      = 2;   // 预财富值
    string  useable_cash  = 3;   // 可以提取的财富值
    string  fetched_cash  = 4;   // 已经提取的财富值
    int32   rank          = 5;   // 资产排行
    string  desc          = 6;   // 描述
    int32   transid       = 32;   // 标示请求
}

// 同步数据的ack
message AckRequest
{
    int32   type          = 1;
    repeated int64   id            = 2;
    int32   time          = 3;
    int32   transid       = 32;   // 标示请求
}


// 存储任务
message SaveTaskRequest
{
    repeated TaskInfo taskinfo     = 1;
    int32   transid       = 32;   // 标示请求
}

message SaveTaskResult
{
    RetBase retbase       = 1;
    int32   transid       = 32;   // 标示请求
}


// 提现协议
message WithdrawRequest
{
    int32   type          = 1;   // 提现类型，微信，qq，支付宝
    int32   withdraw_cash = 2;   // 提现金额，单位是分
    string  id            = 3;   // 客户端生成的提现id
    string  open_id       = 4;   // 客户端第三方认证获得的openid
    string  passwd        = 5;   // 账户密码
    string  user_name     = 6;   // 用户名，真实名字
    string  desc          = 7;   // 提现描述
    int32   time          = 8;   // 请求的时间
    int32   transid       = 32;   // 标示请求
}

message WithdrawResult
{
    RetBase retbase       = 1;
    int32   type          = 2;   // 提现类型，微信，qq，支付宝
    int32   withdraw_cash = 3;   // 提现金额，单位是分
    string  trade_id      = 4;   // 提现时服务端生成的交易id
    string  desc          = 5;   // 提现描述
    int32   time          = 6;   // 请求的时间
    string  pre_cash      = 7;   // 账户余额
    string  useable_cash  = 8;   // 账户可提现的余额，useable_cash < pre_cash
    string  fetched_cash  = 9;   // 已提现的金额
    int32   transid       = 32;   // 标示请求
}


// mysql相关
message MysqlRequest
{
    int32   sqlid      = 1;
    string  sql        = 2;
    int32   isrequestfield = 3;
    int32   limit      = 4;
    int32   timeout    = 5;
}

message MysqlResult
{
    RetBase retbase       = 1;
    int32   affectedrows  = 2;
    int32   insertid      = 3;
    repeated PRow    rows          = 4;
    int32   sqlid         = 5;
}

// redis proxy相关
message RedispRequest
{
    int32   sqlid      = 1;
    string  key        = 2;
    int32   param      = 3;
    int32   limit      = 4;
    int32   timeout    = 5;
    int32   transid       = 32;   // 标示请求
}

message RedispResult
{
    RetBase retbase       = 1;
    int32   affectedrows  = 2;
    int32   insertid      = 3;
    repeated PRow    rows          = 4;
    int32   sqlid         = 5;
    int32   transid       = 32;   // 标示请求
}


// 服务器内部通用的协议格式
message InnerCommonMsg 
{
    repeated PDataCell items       = 1;
    string    transfer    = 2;   // 标示透传的内容
    int32     time        = 3;   // 标示时间
    int32     transid     = 32;  // 标示请求
}